src/platform/EFR32/PlatformManagerImpl.cpp

```diff
@@ -45,6 +45,9 @@ CHIP_ERROR PlatformManagerImpl::_InitChipStack(void)
     // Initialize LwIP.
     tcpip_init(NULL, NULL);
 
+    // Init Chip memory management before the stack
+    chip::Platform::MemoryInit();
```

Comment:
Seems fine, but shouldn't there be a `chip::Platform::MemoryShutdown()` somewhere too?

Reply:
is there a case where we would want to deinit this in a application context?

I could add it here 
```
template <class ImplClass>
CHIP_ERROR GenericPlatformManagerImpl_FreeRTOS<ImplClass>::_Shutdown(void)
{
    return CHIP_ERROR_NOT_IMPLEMENTED;
}
```

Or implement a Shutdown function in PlatformManagerImpl.ccp. Currently the shutdown isn't used nor implemented i seems

Reply:
I'm not actually sure what the intended semantics of `MemoryShutdown` are....

Reply:
I don't know the history, but my understanding is that it's meant to be able to release any unFree'd memory in an environment where that doesn't otherwise happen on exit. It would also be the place to reset any configuration-specific actions associated with MemoryInit(), e.g. mbedtls_platform_set_calloc_free() when mbedtls is used. And it's also the natural place for an instrumented allocator to report results.

(My personal opinion is that CHIPMem.h is is strictly inferior to supplying the desired allocator through the normal malloc API using normal library mechanisms.)

Reply:
Currently CHIP memory management is set as standard malloc/free apis
```
  # Memory management style: malloc, simple, platform.
  chip_config_memory_management = "malloc"
```

In that case, MemoryInit and MemoryShutdown do not do much except checking on themselves and making sure they have been called and called only once
```
static void VerifyInitialized(const char * func)
{
    if (!memoryInitialized)
    {
        fprintf(stderr, "ABORT: chip::Platform::%s() called before chip::Platform::MemoryInit()\n", func);
        abort();
    }
}

#endif

CHIP_ERROR MemoryInit(void * buf, size_t bufSize)
{
#ifndef NDEBUG
    if (memoryInitialized++ > 0)
    {
        fprintf(stderr, "ABORT: chip::Platform::MemoryInit() called twice.\n");
        abort();
    }
#endif
    return CHIP_NO_ERROR;
}

void MemoryShutdown()
{
#ifndef NDEBUG
    if (--memoryInitialized < 0)
    {
        fprintf(stderr, "ABORT: chip::Platform::MemoryShutdown() called twice.\n");
        abort();
    }
#endif
}
```

If chip_config_memory_management was set to something else it would be different.

I can add a PlatformShutdown() in PlatformManagerImpl.cpp where MemoryShutdown() would be call and any other eventual needed denit can be added.

It that good with you?